src/ressa/context.rs

```diff
@@ -99,24 +111,22 @@ impl ContextObjectActions for ParserContext {
     }
 
     fn resolve_tag(&self, name: &str) -> String {
-        if let Some(map) = &self.variables.get(name).as_ref() {
+        // Check that the object exists
+        if let Some(map) = self.objectlike_data.get(name) {
+            // Check that the object is a tag
             if map.contains_key(RESOLVES_TO) {
-                if &*map
-                    .get(RESOLVES_TO)
-                    .unwrap()
-                    .as_ref()
-                    .expect("RESOLVES_TO malformed, does not have a value")
-                    == name
-                {
-                    panic!("{}", name);
+                // Check that the tag is not self-referential, and return resolving it
+                if let Value::String(tag) = map.get(RESOLVES_TO).expect("Contains key lied") {
```

Comment:
I think we can get rid of the `contains_key` call and change this to
```rust
if let Some(Value::String(tag) = map.get(RESOLVES_TO) { /* ... */ }
```

Reply:
Did not think of that; changed.